osu.Desktop/OsuGameDesktop.cs

```diff
@@ -117,9 +117,22 @@ private void fileDrop(object sender, FileDropEventArgs e)
         /// </summary>
         private class StableStorage : WindowsStorage
         {
+            private const string default_songs_path = "Songs";
+
+            private string songsPath;
+            private bool usesCustomSongsPath;
```

Comment:
Why do `songsPath` and `usesCustomSongsPath` need to both exist? Can't we just start off with `songsPath == default_songs_path`, then set value of `songsPath` to the custom one if it turns out that it is different?